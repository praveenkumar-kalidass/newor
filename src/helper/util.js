import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
