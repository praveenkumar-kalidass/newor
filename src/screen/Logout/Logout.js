import React, { useCallback, useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import {
  Flex, Text, Center, Image, useToast,
} from 'native-base';
import AsyncStorage from '@react-native-async-storage/async-storage';

import CONSTANT from 'constant';
import ROUTE from 'constant/route';
import NEWOR_VERIFY from 'asset/image/newor.png';
import Translation from 'translation/Translation';
import AppButton from 'component/AppButton';
import ToastAlert from 'component/ToastAlert';
import useUserApi from 'api/useUser';
import useUser from 'provider/User/useUser';
import useTranslation from 'translation/useTranslation';
import { Container } from './Logout.style';

const Logout = () => {
  const [isSubmit, setIsSubmit] = useState(false);
  const navigation = useNavigation();
  const { logout } = useUserApi();
  const { setUser, setIsAuthorized } = useUser();
  const { translate } = useTranslation();
  const toast = useToast();

  const doLogout = useCallback(async () => {
    try {
      setIsSubmit(true);
      await logout();
      setUser({});
      await AsyncStorage.removeItem(CONSTANT.STORAGE_KEY.TOKEN);
      setIsAuthorized(false);
      navigation.navigate(ROUTE.LOGIN);
      setIsSubmit(false);
    } catch (error) {
      setIsSubmit(false);
      let errorMessage = translate('ERROR_CODE.NEWOR_INTERNAL_SERVER_ERROR');
      const errorCode = error?.response?.data?.code;
      if (errorCode) {
        errorMessage = translate(`ERROR_CODE.${errorCode}`);
      }
      toast.show({
        render: () => <ToastAlert status="error" message={errorMessage} />,
        placement: 'bottom',
      });
    }
  }, [navigation]);

  return (
    <Flex flex={1} direction="row" p="5" justify="center">
      <Container spacing={5}>
        <Center><Image alt={CONSTANT.APP_NAME} source={NEWOR_VERIFY} size="2xl" /></Center>
        <Translation tkey="SURE_TO_LOGOUT" as={Text} textAlign="center" fontSize="lg" />
        <Translation
          tkey="YES"
          as={AppButton}
          variant="primary"
          onPress={doLogout}
          isLoading={isSubmit}
          testID="logout-yes"
        />
        <Translation
          tkey="NO"
          as={AppButton}
          variant="secondary"
          onPress={navigation.goBack}
          isDisabled={isSubmit}
          testID="logout-no"
        />
      </Container>
    </Flex>
  );
};

export default Logout;
