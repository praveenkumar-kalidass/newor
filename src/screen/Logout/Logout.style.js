import styled from 'styled-components/native';
import { VStack } from 'native-base';

// eslint-disable-next-line import/prefer-default-export
export const Container = styled(VStack).attrs(() => ({
  flex: 1,
  alignSelf: 'center',
  p: 5,
  space: 5,
}))``;
