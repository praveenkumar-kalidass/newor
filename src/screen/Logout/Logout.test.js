import React from 'react';

import { render, fireEvent, act } from 'test/util';
import Logout from './Logout';

const mockNavigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};
jest.mock('@react-navigation/native', () => {
  const reactNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...reactNavigation,
    useNavigation: () => mockNavigation,
  };
});
const mockUseUser = {
  logout: jest.fn(),
};
jest.mock('api/useUser', () => () => mockUseUser);
const mockToast = { show: jest.fn() };
jest.mock('native-base', () => {
  const nativeBase = jest.requireActual('native-base');
  return {
    ...nativeBase,
    useToast: () => mockToast,
  };
});

describe('Logout', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const container = render(<Logout />);

    expect(container).toMatchSnapshot();
  });

  it('should go back on declining logout', () => {
    const { getByTestId } = render(<Logout />);

    fireEvent.press(getByTestId('logout-no'));

    expect(mockNavigation.goBack).toHaveBeenCalledTimes(1);
  });

  it('should go to login on approving logout', async () => {
    mockUseUser.logout.mockResolvedValueOnce();

    const { getByTestId } = render(<Logout />);

    await act(async () => {
      await fireEvent.press(getByTestId('logout-yes'));
    });

    expect(mockNavigation.navigate).toHaveBeenCalledWith('LOGIN');
  });

  it('should show toast on failed logout', async () => {
    mockUseUser.logout.mockRejectedValueOnce();

    const { getByTestId } = render(<Logout />);

    await act(async () => {
      await fireEvent.press(getByTestId('logout-yes'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(1);
  });
});
