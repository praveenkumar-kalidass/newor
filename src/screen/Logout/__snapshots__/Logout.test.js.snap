// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
          }
        }
      >
        <View
          dataSet={Object {}}
          spacing={5}
          style={
            Array [
              Object {
                "alignSelf": "center",
                "flex": 1,
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <Image
              accessibilityLabel="newor"
              alt="newor"
              dataSet={Object {}}
              onError={[Function]}
              source={
                Object {
                  "testUri": "../../../src/asset/image/newor.png",
                }
              }
              style={
                Object {
                  "height": 256,
                  "maxWidth": "100%",
                  "width": 256,
                }
              }
            />
          </View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "height": 20,
              }
            }
          />
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#171717",
                "fontFamily": undefined,
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 27,
                "textAlign": "center",
                "textDecorationLine": undefined,
              }
            }
          >
            SURE_TO_LOGOUT
          </Text>
          <View
            dataSet={Object {}}
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            colorScheme="primary"
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ad0000",
                "borderRadius": 4,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 10,
              }
            }
            testID="logout-yes"
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              test={true}
            >
              <View
                dataSet={Object {}}
                style={Object {}}
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#fafafa",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  YES
                </Text>
              </View>
            </View>
          </View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            colorScheme="primary"
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00238a",
                "borderRadius": 4,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 10,
              }
            }
            testID="logout-no"
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              test={true}
            >
              <View
                dataSet={Object {}}
                style={Object {}}
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#fafafa",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  NO
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          style={
            Array [
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#7f7f7f",
                "paddingBottom": 8,
                "paddingLeft": 16,
                "paddingRight": 20,
                "paddingTop": 8,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#a1a1aa",
                    "borderColor": "#27272a",
                    "borderRadius": 9999,
                    "height": 96,
                    "justifyContent": "center",
                    "position": "relative",
                    "width": 96,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 40,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                dataSet={Object {}}
                style={Object {}}
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "marginBottom": 8,
                    "marginTop": 8,
                    "textDecorationLine": undefined,
                  }
                }
              >
                undefined undefined
              </Text>
            </View>
          </View>
        </View>
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
                "onPress": [Function],
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
